### DISCUSSION TITLE
go2rtc help for Wyze Cam3

### DISCUSSION THREAD

**Discussion by teredactle** [gtxaspec/wz_mini_hacks#601] [Q&A]
Hi,
I'm looking to try go2rtc on my WC3 but I'm a bit lost on what I need in the YML file. Going to go2rtc github looking at the further instruction doesn't help (not sure where to find an example even).

Just looking to get RTSP working as from what I'm reading go2rtc is more stable (_last time I tried mini hacks this options was not available_), and offers better audio - I'm running the wyze RTSP beta on the wc3 and audio is horrible.

Does the camera need static IP, I looked as other examples and the config file show IPs. Just looking for simple RTSP feeds with better audio, at 20FPS is fine.... can't find any examples in particular for this, perhaps someone can point me in the right direction. 

Is it just easier to add a stream by going to http://<camera_ip>:1984 ?  Haven't tried, will give it a go tomorrow.

Thanks

**Comment by MyCodeRocks** [comment]
Does the camera need static IP - No it doesn't you should for better reliability i.e. but do this on your networking device (like your switch/router etc), I have mine working on DHCP fine but to be sure that the rtsp stream doesn't fail I set static IPs eventually on my router.

Just looking for simple RTSP feeds with better audio -- So insure your config has go2rtc ONLY as a rtsp service running, then boot the camera go to http://<IP_Address>:1984 see if the streams are showing? 

if they showing you will see a "links" link next to the stream click on that, it will list all the available links - they just copy and paste the output you want to something like VLC or Blueiris.

Personally what I did to make sure it was working:
1. I set rtsp authentication for now off.
2. I then used the link as per above in VLC - Network stream - made sure it was coming though.
3. Then I added it to my BlueIris install and works.

What I have noticed - Audio works and is clear but is about 3-5 seconds delayed.

**Reply by teredactle** [reply]
Thanks, I will give this a go when I have a chance! Glad to read this :)
- the 3-5s delay can be adjusted directly in Blue Iris, so that the stream is recorded properly, unless this changes over time (which I read somewhere, but I think it was on Wyze RTSP implementation).

**Reply by teredactle** [reply]
Having an issue, running the w3 beta fw and with mini-hacks can't get it to register to wifi... if I take the SD card out it registers just fine...

Using the POE adapter, it works fine over the LAN, however, if I unplug that and just power the camera with a micro USB it will not register to the wifi...

**Comment by bjs-pdx** [comment]
"So insure your config has go2rtc ONLY as a rtsp service running,"
How do you set that in the config?  There are settings in the YAML file for different streams (1080p vs 360p) but I don't see any setting for RTSP.

**Reply by teredactle** [reply]
Just disable RTSP on the other (standard) section.

**Reply by bjs-pdx** [reply]
I don't have RTSP enabled in the config file, only go2rtc.  How do I access the RTSP feed from go2rtc?

**Reply by gtxaspec** [reply]
@bjs-pdx on the :1984 page,:
![image](https://github.com/gtxaspec/wz_mini_hacks/assets/12115272/94f40c5b-aecd-442b-b9e9-9b3d022a68e5)
click on "Links" for the stream that you would like to access
these are your links to access rtsp:
![image](https://github.com/gtxaspec/wz_mini_hacks/assets/12115272/cdc60316-8944-4569-9c8d-0ef5e043a8c4)

**Reply by bjs-pdx** [reply]
So, for VLC, what network stream would I connect to for rtsp?  rtsp://192.168.x.x:8554

**Reply by gtxaspec** [reply]
Right click on the rtsp links and copy whatever URL it provides, that's your link to paste into vlc

**Reply by bjs-pdx** [reply]
Tried that and it doesn't work:

http://192.168.x.x:1984/stream.html?src=1080p&mode=webrtc,mse,hls,mjpeg

**Reply by gtxaspec** [reply]
Not that one, just the links that say rtsp, reference the second photo above

**Reply by bjs-pdx** [reply]
There aren't any.

![image](https://github.com/gtxaspec/wz_mini_hacks/assets/14082937/73d6c9cf-e841-4262-b81f-5195de462b8b)

**Reply by gtxaspec** [reply]
Click on "links"

**Reply by bjs-pdx** [reply]
I did and when I do my browser goes to the link I posted first.

**Reply by gtxaspec** [reply]
Strange try the link directly http://[IP]:1984/links.html?src=1080p

**Reply by bjs-pdx** [reply]
I get this:

![image](https://github.com/gtxaspec/wz_mini_hacks/assets/14082937/aae22fae-eca6-42ef-a367-bf1a8d616f44)

**Reply by gtxaspec** [reply]
The rtsp links should work for you

**Reply by bjs-pdx** [reply]
Great! That's what I was trying to find.  Not very obvious in any of the documentation.

rtsp://192.168.x.x:8554/1080p

Thank you!

**Comment by alej908** [comment]
With the go2rtc, can the frame rate and quality be changed? Is it a concern for there to not be authentication if the cameras are blocked from the internet and are in self hosted mode?  Technically, someone would need access to a local network and then sniff for feeds for there to be an issue right?

**Reply by gtxaspec** [reply]
You can always enable authentication via go2rtc https://github.com/AlexxIT/go2rtc#module-rtsp

Framerate and quality can not be changed in go2rtc, unless you enable transcoding via ffmpeg, otherwise you can change the framerate and quality via wz_mini config. 
https://github.com/gtxaspec/wz_mini_hacks/wiki/Configuration-File#video-settings

**Reply by alej908** [reply]
Thank you! I missed that.

Also, the switch to go2rtc has been amazing. No more artifacts in Frigate or half of the image going away. I have 5 v3s all currently on wifi. The mini hacks was great, btu this takes it up several notches. I have all the hardware to have them wired and usb to ethernet. Just need time.

**Reply by saihgupr** [reply]
@alej908 What settings did you change to get those results?

**Comment by noloboy** [comment]
Hello, I'm trying out the go2rtc links which works with excellent video but there is no audio at all for me.  Tested all "links" options and none of them gave sound from the camera.

If I use just the regular RTSP from WZ_Mini, there is audio.

Did you have to put additional configuration in the go2rtc yml file to produce sound?

**Reply by epvuc** [reply]
I have a Wyze Cam v3 running 4.36.11.7071.   go2rtc is configured and running, v4l2rtspserver is not enabled.  When anything tries to start a stream (or even when I run ffmpeg by hand from the shell on the camera)  ffmpeg fails with 

```
[video4linux2,v4l2 @ 0x77166820] ioctl(VIDIOC_G_FMT): Invalid argument
/dev/video1: Invalid argument
```
There's also /dev/video2 which fails the same way. I feel like something weird is going on if VIDIOC_G_FMT doesn't even work. lsof doesn't show any other process holding /dev/video1 open but that doesn't mean much.   The camera is working fine and is viewable via the Wyze app from my phone.  ffmpeg can access audio fine. It's just video.

Similarly, if I start v4l2rtspserver instead, it fails on /dev/video1 in the same way: 

```
[NOTICE] /root/CLEAN/v4l2rtspserver/main.cpp:297
	Version: 0.3.3-4-ga43e094-dirty live555 version:2022.04.26
[NOTICE] /root/CLEAN/v4l2rtspserver/src/V4l2RTSPServer.cpp:37
	Create V4L2 Source.../dev/video1
[NOTICE] /root/CLEAN/v4l2rtspserver/v4l2wrapper/src/V4l2Device.cpp:133
	driver:v4l2 loopback capabilities:85008003 mandatory:4000001
[NOTICE] /root/CLEAN/v4l2rtspserver/v4l2wrapper/src/V4l2Device.cpp:135
	/dev/video1 support output
[NOTICE] /root/CLEAN/v4l2rtspserver/v4l2wrapper/src/V4l2Device.cpp:136
	/dev/video1 support capture
[NOTICE] /root/CLEAN/v4l2rtspserver/v4l2wrapper/src/V4l2Device.cpp:138
	/dev/video1 support read/write
[NOTICE] /root/CLEAN/v4l2rtspserver/v4l2wrapper/src/V4l2Device.cpp:139
	/dev/video1 support streaming
[ERROR] /root/CLEAN/v4l2rtspserver/v4l2wrapper/src/V4l2Device.cpp:192
	/dev/video1: Cannot get format Invalid argument
[ERROR] /root/CLEAN/v4l2rtspserver/v4l2wrapper/src/V4l2Device.cpp:192
	/dev/video1: Cannot get format Invalid argument
[ERROR] /root/CLEAN/v4l2rtspserver/v4l2wrapper/src/V4l2Device.cpp:192
	/dev/video1: Cannot get format Invalid argument
[ERROR] /root/CLEAN/v4l2rtspserver/v4l2wrapper/src/V4l2Device.cpp:83
	Cannot init device:/dev/video1
[NOTICE] /root/CLEAN/v4l2rtspserver/v4l2wrapper/src/V4l2MmapDevice.cpp:141
	Device /dev/video1VIDIOC_STREAMOFF: Bad file descriptor
VIDIOC_REQBUFS: Bad file descriptor
```

So it looks like something is up with the kernel driver for /dev/videoN rather than with either of the userland rtsp servers. 

I don't have a toolchain set up to bulid the v4l2 utilities to check /dev/video* directly but neither ffmpeg nor anything else installed on the camera can do anything with it.  Any ideas?   I'm probably overlooking something stupid because I just found wz_mini_hacks a couple hours ago but I'm very excited about its existence!

Any ideas?

**Reply by archandanime** [reply]
That version is not supported by wz_mini_hacks, you need to rollback to `4.36.9.139` to use video streaming:
https://github.com/gtxaspec/wz_mini_hacks/wiki/Firmware-Support
Reason: https://github.com/gtxaspec/wz_mini_hacks/issues/187

**Comment by epvuc** [comment]
> That version is not supported by wz_mini_hacks, you need to rollback to `4.36.9.139` to use video streaming: https://github.com/gtxaspec/wz_mini_hacks/wiki/Firmware-Support Reason: #187

That was indeed the problem, thanks! Everything looks good now, go2rtc works very well.

**Comment by frankkubler** [comment]
hi, I just saw this thread, after searching in this github repo, I don't found any information on how to rollback.
Someone have this information ?
Many thanks

**Reply by Bob-AOL** [reply]
https://forums.wyze.com/t/auto-update-failure-how-to-rollback-wyze-cam-v3-pro-firmware-or-similar/279998

Not a very good resource. Sorry.
